contracts/duckies/ducklings/Ducklings.sol

```diff
@@ -0,0 +1,505 @@
+		_setDefaultRoyalty(msg.sender, ROYALTY_FEE);
+
+		// TODO: define price
+		mintPrice = 10_000 * 10 ** duckiesContract.decimals();
+		meldPrice = 10_000 * 10 ** duckiesContract.decimals();
+
+		meldWeights = [6, 5, 4, 3, 2, 1];
+		meldingZombeakWeights = [[10, 90], [5, 90], [2, 90]];
+
+		duckiesContract = ERC20BurnableUpgradeable(ducklingsAddress);
+	}
+
+	// -------- Upgrades --------
+
+	function _authorizeUpgrade(
+		address newImplementation
+	) internal override onlyRole(UPGRADER_ROLE) {}
+
+	// -------- ERC721 --------
+
+	function _burn(uint256 tokenId) internal override(ERC721RoyaltyUpgradeable, ERC721Upgradeable) {
+		super._burn(tokenId);
+	}
+
+	function tokenURI(
+		uint256 tokenId
+	) public view override(ERC721Upgradeable) returns (string memory) {
+		Duckling memory duckling = tokenIdToDuckling[tokenId];
+
+		return
+			bytes(apiBaseURL).length > 0
+				? string(
+					abi.encodePacked(
+						apiBaseURL,
+						duckling.gene.toString(),
+						'-',
+						uint256(duckling.birthdate).toString(),
+						'-',
+						uint256(duckling.collectionId).toString()
+					)
+				)
+				: '';
+	}
+
+	function supportsInterface(
+		bytes4 interfaceId
+	)
+		public
+		view
+		virtual
+		override(ERC721RoyaltyUpgradeable, ERC721Upgradeable, AccessControlUpgradeable)
+		returns (bool)
+	{
+		return super.supportsInterface(interfaceId);
+	}
+
+	// -------- ERC2981 Royalties --------
+
+	function setRoyaltyCollector(address account) public onlyRole(DEFAULT_ADMIN_ROLE) {
+		_royaltiesCollector = account;
+	}
+
+	function getRoyaltyCollector() public view returns (address) {
+		return _royaltiesCollector;
+	}
+
+	// -------- API URL, Price --------
+
+	function setAPIBaseURL(string calldata apiBaseURL_) external onlyRole(MAINTAINER_ROLE) {
+		apiBaseURL = apiBaseURL_;
+	}
+
+	function setMintPrice(uint256 price) external onlyRole(MAINTAINER_ROLE) {
+		mintPrice = price;
+	}
+
+	function setMeldPrice(uint256 price) external onlyRole(MAINTAINER_ROLE) {
+		meldPrice = price;
+	}
+
+	// -------- Collections --------
+
+	function addCollection(Collection calldata collection) external onlyRole(MAINTAINER_ROLE) {
+		setCollection(collection, uint8(nextCollectionId.current()));
+		nextCollectionId.increment();
+	}
+
+	function setCollection(
+		Collection calldata collection,
+		uint8 collectionId
+	) public onlyRole(MAINTAINER_ROLE) {
+		if (
+			collection.availableBefore <= block.timestamp ||
+			collection.traitWeights.length != uint8(type(Gene.Traits).max) + 1
+		) revert InvalidCollection(collection);
+
+		collectionOfId[collectionId] = collection;
+	}
+
+	function obsoleteCollection(uint8 collectionId) external onlyRole(MAINTAINER_ROLE) {
+		_requireValidCollection(collectionId);
+		collectionOfId[collectionId].availableBefore = uint64(block.timestamp);
+	}
+
+	// -------- Mint, Meld --------
+
+	function mintByPayment(uint8 collectionId, uint8 amount) external UseRandom {
+		duckiesContract.burnFrom(msg.sender, mintPrice * amount);
```

Comment:
We are not burning funds, we are collecting to Treasury

Reply:
oh . good